#! /usr/bin/env python
#Autogenerated V4L Dynamic Control

PACKAGE='tuw_checkerboard'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("show_camera_image", bool_t, 0, "Show camera image", True)
gen.add("show_camera_image_waitkey", int_t, 0, "Time to draw the camera image in ms", 2, 1, 1000)
gen.add("input_raw", bool_t, 0, "the imput image is distored", True)
gen.add("rotate_camera_image_180", bool_t, 0, "Rotate input camera image by 180 deg", False)
gen.add("plubish_tf", bool_t, 0, "publish tf messages", True)
gen.add("plubish_marker", bool_t, 0, "plublish marker messages", True)
gen.add("publish_fiducials", bool_t, 0, "publish fiducials to allow 3th party pose estimation.", False)
gen.add("publish_pose", bool_t, 0, "publish checkerboard pose", False)
gen.add("checkerboard_columns", int_t, 0, "interior number of colomn corners", 8, 1, 20)
gen.add("checkerboard_rows"   , int_t, 0, "interior number of row corners"   , 6, 1, 20)
gen.add("checkerboard_square_size", double_t, 0, "checkerboard square size [m]", 0.03, 0, 1)
gen.add("checkerboard_min_square_size", int_t, 0, "minimal checkerboard square size in [pix] to accept a detection", 10, 1, 100)
gen.add("adaptive_thresh", bool_t, 0, "checkerboard detection flags", True)
gen.add("normalize_image", bool_t, 0, "checkerboard detection flags", False)
gen.add("filter_quads", bool_t, 0, "checkerboard detection flags", False)
gen.add("fast_check", bool_t, 0, "checkerboard detection flags", False)
gen.add("subpixelfit", bool_t, 0, "Sub-pixel accurate corner locator", True)
gen.add("subpixelfit_window_size", int_t, 0, "Half of the side length of the search window [pix]", 8, 1, 20)

exit(gen.generate(PACKAGE, "tuw_checkerboard", "CheckerboardDetection"))
